znoble360/MyThirdEar
# Added Hive DB, fixed caching issues

### Code Additions

- Users can now upload files, exit out of the app completely, re open and the files will be there and play correctly using the previously generated files
- Added Hive to use as a key-value DB
  - Created a AudioFileData model to hold the audio file metadata and paths in the DB
- Implemented the auto-population of the Library model with AudioFile objects that are fetched from Hive and generated when the user opens the app

### How to test this
***NOTE**: For some reason, when you upload a file, if you click it immediately after, sometimes the play button stays loading. If you instead, upload the file and wait 2-3 seconds before clicking it, it doesn't happen. I'm guessing there's some loading going on that we're not waiting for. I'll make an issue for it soon.*

- First, make sure you delete the app on the emulator and reinstall before running **(YOU ONLY NEED TO RE-INSTALL ONCE, so that the DB and file stuff starts from scratch)**
- Run the app using `flutter run --no-sound-null-safety`
- Try out adding files and playing them. Then end the program and re run, and make sure the files are still there and playable.
